### PR TITLE
Splink dataframe docs

### LINKED ISSUE #1428: Some `SplinkDataFrame` methods are undocumented
See https://github.com/moj-analytical-services/splink/discussions/1421#discussioncomment-6404664.

Most `SplinkDataFrame` methods are undocumented - neither docstrings nor on docs site - these should be added.

### PR DESCRIPTION
**Pull request by ADBond**
### Type of PR

- [ ] BUG
- [ ] FEAT
- [ ] MAINT
- [x] DOC

### Is your Pull Request linked to an existing Issue or Pull Request?

Closes #1428.

### Give a brief description for the solution you have provided

Added docstrings to methods lacking in `SplinkDataFrame`, and added these methods to API page. Expanded / updated topic guide on querying using these.

Additionally have made a couple of code changes - these are just **renamings** for consistency across backends:
* `DuckDBLinkerDataFrame` -> `DuckDBDataFrame`
* `SplinkDataframe` -> `SplinkDataFrame`

Also fixed a rendering issue with `linker.deterministic_link()`

### PR Checklist

- [x] Added documentation for changes
- [ ] Added feature to example notebooks at tutorial in [splink_demos](https://github.com/moj-analytical-services/splink_demos) (if appropriate)
- [ ] Added tests (if appropriate)
- [ ] Made changes based off the latest version of Splink
- [ ] Run the [linter](https://moj-analytical-services.github.io/splink/dev_guides/changing_splink/lint.html)

### CODE REVIEW

File: docs/topic_guides/splink_fundamentals/querying_splink_results.md [hunk 1]
```diff
@@ -1,26 +1,42 @@
...
+tags:
+  - SQL
+  - Data Frames
+  - SplinkDataFrame
+toc_depth: 2
+---
+# Retrieving and Querying Splink Results
 
-Splink returns tables of results using a class called a `SplinkDataFrame`. e.g. when you run `df_predict = linker.predict()` `df_predict` is a `SplinkDataFrame`
+Once you have [created your linkage model](./settings.md), [trained your parameters](../demos/tutorials/04_Estimating_model_parameters.html) and [predicted your match probabilities](../demos/tutorials/05_Predicting_results.html) you may want to run queries on your results. Splink provides several methods for interacting with the tables produced in this process.
```
Comment (RossKen) [inline docs/topic_guides/splink_fundamentals/querying_splink_results.md:10]: demos links should be e.g.
`../../demos/tutorials/04_Estimating_model_parameters.html`

**Review by RossKen — CHANGES_REQUESTED** [review]
Hey @ADBond this is brilliant! Thanks for putting it together 👏

Just one thing from me - some of the links that refer to directories outside of `topic_guides` aren't working. I have left a comment on one example. In most cases, it is just figuring out how many levels to go up with `../`.

File: docs/topic_guides/splink_fundamentals/querying_splink_results.md [hunk 2]
```diff
@@ -32,14 +48,75 @@ from {results.physical_name}
...
+# or alternatively:
+df_predict.to_csv("splink_predictions.csv", overwrite=True)
 ```
 
-Note that `linker.query_sql` will return a pandas dataframe by default, but you can instead return a SplinkDataFrame as follows:
+### Creating a `SplinkDataFrame`
 
+Generally speaking, any Splink method that results in a table will return a `SplinkDataFrame`. This includes:
+
+* [`linker.predict()`](../linker.md#splink.linker.Linker.predict)
```
Comment (RossKen) [inline docs/topic_guides/splink_fundamentals/querying_splink_results.md:80]: This section of links also need updated

**Comment by ADBond** [comment]
> Hey @ADBond this is brilliant! Thanks for putting it together 👏
> 
> Just one thing from me - some of the links that refer to directories outside of `topic_guides` aren't working. I have left a comment on one example. In most cases, it is just figuring out how many levels to go up with `../`.

Ah thanks - think these got messed up after I merged `master` in. These should all be fixed now.

**Comment by RossKen** [comment]
Super - thank you!

Sorry, one more thing - would you mind adding an example to the docstrings to each of the SplinkDataframe functions so that there are easy copy-paste code for users?

**Pull request merged**